interface/src/app/node/node.js

```diff
@@ -304,6 +307,32 @@ angular.module('adage.node', [
         $scope.numExpShown =
           $scope.topMode ? $scope.topNum : $scope.experiments.length;
       };
+
+      // Event handler when user clicks the checkbox in front of each sample.
+      // It keeps track of the number of selected samples in each experiment
+      // to indicate whether the "show annotations" button should be enabled.
+      $scope.toggleSelection = function(exp, selected) {
+        if (selected) {
+          ++exp.numSelections;
+        } else {
+          --exp.numSelections;
+        }
+      };
```

Comment:
On a slightly related topic - I think that this large ``highRangeExp`` directive can actually also be divided into smaller directives/components, where one is the directive/component for the Node page quadrant that holds all of the Highest Range Experiments, and it calls smaller directives/components for each experiment itself. That way, the experiment in each component only knows information about itself and the samples it contains. However, this would almost surely fall outside the scope of this Pull Request, but it is just something I was also thinking about.